pkgs/tools/misc/hdf4/default.nix

```diff
@@ -31,11 +31,12 @@ stdenv.mkDerivation rec {
     "-DHDF4_BUILD_WITH_INSTALL_NAME=OFF"
     "-DHDF4_ENABLE_JPEG_LIB_SUPPORT=ON"
     "-DHDF4_ENABLE_NETCDF=OFF"
-    "-DHDF4_ENABLE_SZIP_ENCODING=ON"
-    "-DHDF4_ENABLE_SZIP_SUPPORT=ON"
     "-DHDF4_ENABLE_Z_LIB_SUPPORT=ON"
     "-DHDF4_BUILD_FORTRAN=OFF"
     "-DJPEG_DIR=${libjpeg}"
+  ] ++ stdenv.lib.optionals (szip != null) [
+    "-DHDF4_ENABLE_SZIP_ENCODING=ON"
```

Comment:
@phile314-fh I don't subscribe to the idea that the rest of nixpkgs has reached perfection and I absolutely hate comparison arguments, because they are almost always wrong and uninformed. Your version is particularly annoying, because you just hope you grepped enough. So, instead of just implementing exactly what I asked for (which is just two conditionals(!)), you thought it was a better idea to ignore my lengthy and extremely detailed response and basically tell me to go fuck myself.

If your goal was for me to like you, you failed miserably.